prestopantry_app/tests/view_tests/test_my_pantry_page_view.py

```diff
@@ -0,0 +1,37 @@
+from django.test import TestCase
+from django.urls import reverse
+from prestopantry_app.models.users import User
+from prestopantry_app.models.user_ingredients import UserIngredient
+
+
+class MyPantryPageViewTest(TestCase):
+    def setUp(self):
+        self.user = User.objects.create_user('goldmember','drevil@gmail.com','ilovegold')
+
+    def test_my_pantry_view(self):
+        self.client.force_login(self.user)
+        response = self.client.get('/pantry/')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'pantry.html')
+
+    def test_my_pantry_ingredients(self):
+         # login
+        self.client.force_login(self.user)
+        # go to pantry search page
+        response = self.client.get(reverse('pantry'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'pantry.html')
+        self.assertContains(response, '<h2> My Pantry </h2>')
+
+        # test search
+        data = {'q': '88'}
+        response = self.client.get(reverse('pantry'), data)
+        self.assertContains(response, 'Nothing in pantry')
+
+        session = self.client.session
+        session.update({'ingredient_search_results': {'ingredient_info': [['Torkelson Cheese Co. Brick Cheese Wisconsin', 406181, 123344564378]]}})
+        session.save()
+        data = {'ingredient_name': 'Torkelson Cheese Co. Brick Cheese Wisconsin',
+                'ingredient_id': '406181', 'add_ingredient_button': '', 'upc': '123344564378'}
+        response = self.client.get('/pantry/', data)
+        self.assertEqual(response.status_code, 200)
```

Comment:
I don't think this is doing what you think it is. You shouldn't be touching session at all. You are verifying that 1) if there are no ingredients on the user it display empty and 2) if there are ingredients on the user it's displaying properly. You should create an ingredient using `UserIngredient.objects.create` and then make sure that is what is in the response.context after making a get request to the page.